src/interfaces/chain.h

```diff
@@ -180,7 +180,7 @@ class Chain
         std::string& err_string) = 0;
 
     //! Calculate mempool ancestor and descendant counts for the given transaction.
-    virtual void getTransactionAncestry(const uint256& txid, size_t& ancestors, size_t& descendants) = 0;
+    virtual void getTransactionAncestry(const uint256& txid, size_t& ancestors, size_t& descendants, size_t* ancestorsize = nullptr, CAmount* ancestorfees = nullptr) = 0;
```

Comment:
Is using `Optional` preferred nowadays for params that can be `{}`?

Reply:
might be a good meeting topic?

Reply:
I think for optional return values it is preferable to pass by pointer. Otherwise the caller code will read clumsy like

```cpp
Optional<int> foo;
GetInt(foo);
std::cout << * foo << std::endl;
```

As opposed to a simple

```cpp
int foo;
GetInt(&foo);
std::cout << foo << std::endl;

Reply:
I wonder whether we should update documentation on L176 too to reflect the change here.